src/User/Controller/AdminController.php

```diff
@@ -346,4 +349,33 @@ public function actionForcePasswordChange($id)
         }
         $this->redirect(['index']);
     }
+
+    /**
+     * Display list session history
+     */
+    public function actionSessionHistory($id)
+    {
+        $searchModel = new SessionHistorySearch([
+            'user_id' => $id,
+        ]);
+        $dataProvider = $searchModel->search(Yii::$app->request->queryParams);
+
+        $user = $this->userQuery->where(['id' => $id])->one();
+
+        return $this->render('_session-history', [
+            'searchModel' => $searchModel,
+            'dataProvider' => $dataProvider,
+            'user' => $user,
+        ]);
+    }
+
+    /**
+     * Terminate all session user
+     */
+    public function actionTerminateSessions($id)
+    {
```

Comment:
Again check if feature is enabled before allowing the action, maybe it should be done at `access` if possible

Reply:
Fix it.